### PR TITLE
Improve guided lessons

### PR DESCRIPTION
**Pull request by ClementPasteau**
Few things:

- some fixes on the existing guided lessons
- more languages for fling game
- some design improvements on small screens
- measure completion and show a completion bar per lesson + global
- show lessons on small screens

Capture d’écran 2023-03-23 à 11 26 08
Capture d’écran 2023-03-23 à 11 26 29
Capture d’écran 2023-03-23 à 11 27 33
Capture d’écran 2023-03-23 à 11 50 02

### CODE REVIEW

File: newIDE/app/src/AssetStore/ExtensionStore/ExtensionsSearchDialog.js [hunk 1]
```diff
@@ -112,7 +114,13 @@ export default function ExtensionsSearchDialog({
               <FlatButton
                 leftIcon={<CloudDownload />}
                 key="import"
-                label={<Trans>Import extension</Trans>}
+                label={
+                  windowWidth === 'small' ? (
```
Comment (4ian) [inline newIDE/app/src/AssetStore/ExtensionStore/ExtensionsSearchDialog.js:118]: Nice!

File: newIDE/app/src/InAppTutorial/InAppTutorialOrchestrator.js [hunk 2]
```diff
@@ -961,6 +971,17 @@ const InAppTutorialOrchestrator = React.forwardRef<
...
                   setDisplayEndDialog(false);
+                  if (isRunningMiniTutorial) {
+                    // If running a mini tutorial, we save the progress to indicate that the user has finished this lesson.
+                    preferences.saveTutorialProgress({
+                      tutorialId: tutorial.id,
+                      userId: authenticatedUser.profile
+                        ? authenticatedUser.profile.id
+                        : undefined,
+                      ...getProgress(),
+                      // We do not specify a storage provider, as we don't need to reload the project.
```
Comment (4ian) [inline newIDE/app/src/InAppTutorial/InAppTutorialOrchestrator.js:982]: Safety check: This won't interfere with fling game?

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/InAppTutorials/InAppTutorialPhaseCard.js [hunk 3]
```diff
@@ -123,7 +123,13 @@ const InAppTutorialPhaseCard = ({
                   <Line noMargin justifyContent="center">
                     <Chip
                       size="small"
-                      label={<Trans>{durationInMinutes} minutes</Trans>}
+                      label={
+                        durationInMinutes === 1 ? (
+                          <Trans>1 minute</Trans>
```
Comment (4ian) [inline newIDE/app/src/MainFrame/EditorContainers/HomePage/InAppTutorials/InAppTutorialPhaseCard.js:128]: Nice

File: newIDE/app/src/Profile/LoginDialog.js [hunk 4]
```diff
@@ -122,7 +123,12 @@ const LoginDialog = ({
             </Text>
           </Link>
         </Column>
-        <div style={styles.formContainer}>
+        <div
+          style={{
+            ...styles.formContainer,
+            width: windowWidth === 'small' ? '100%' : '60%',
```
Comment (4ian) [inline newIDE/app/src/Profile/LoginDialog.js:129]: Nice

File: newIDE/app/src/UI/SortableVirtualizedItemList/index.js [hunk 5]
```diff
@@ -136,6 +138,24 @@ export default class SortableVirtualizedItemList<Item> extends React.Component<
             {screenType => (
               <List
                 ref={list => (this._list = list)}
+                // We override this function to avoid a bug in react-virtualized
+                // where the overscanCellsCount is not taken into account after a scroll
+                // see https://github.com/bvaughn/react-virtualized/issues/1582#issuecomment-785073746
```
Comment (4ian) [inline newIDE/app/src/UI/SortableVirtualizedItemList/index.js:143]: Nice

**Review by 4ian — APPROVED** [review]
Nice!

**Comment by 4ian** [comment]
> Should the bars be replaced when 100% is reached with a "Completed" chip with a cup icon and maybe a golden color? Should the total progress bar disappear when 100% is reached?

We could, I still approved to go ahead with this and we can improve the display later.

**Comment by D5688** [comment]
<details><summary>Details</summary>
<p>

#5146 

</p>
</details>

File: newIDE/app/src/InAppTutorial/InAppTutorialOrchestrator.js (hunk 2 above)
Comment (ClementPasteau) [reply newIDE/app/src/InAppTutorial/InAppTutorialOrchestrator.js:982]: Nope, checked!

**Pull request merged**